snuba/datasets/processors/generic_metrics_processor.py

```diff
@@ -146,10 +156,20 @@ def process_message(
                 message["sentry_received_timestamp"]
             )
 
+        self.__record_cogs(message)
         return InsertBatch(
             [processed], None, sentry_received_timestamp=sentry_received_timestamp
         )
 
+    def __record_cogs(self, message: GenericMetric) -> None:
+        if get_config("enable_gen_metrics_processor_cogs", 0):
+            record_cogs(
+                resource_id=self._resource_id,
+                app_feature=message["use_case_id"],
+                amount=len(json.dumps(message).encode("utf-8")),
```

Comment:
I am not sure how much overhead serializing the message here would cost. You might want to be aware of it when you do the rollout. Maybe make the config percentage based to be safer.

Reply:
pickle is generally serializing faster than JSON.
You can do that.